packages/aws-cdk-lib/aws-ec2/test/vpc.test.ts

```diff
@@ -2261,6 +2261,46 @@ describe('vpc', () => {
 
     });
 
+    test('can filter by multiple CIDR Ranges', () => {
```

Comment:
> Note: It is a bit difficult to write unit tests as CDK doesn't have the ability to add a secondary CIDR to a VPC.

Adding a test case to check that [subnet creation fails](https://github.com/aws/aws-cdk/blob/06bea3159e137a8e85e362795ef2cf7e219e381c/packages/aws-cdk-lib/aws-ec2/lib/vpc-endpoint.ts#L748-L750) with no subnets if non-overlapping CIDR ranges are supplied could help increase coverage.

Reply:
Added additional test case